### PR TITLE
add pending option to elastic:migrate:status command

### PR DESCRIPTION
**Pull request by maximepvrt**
same as `migrate:status --pending` https://github.com/laravel/framework/blob/ca546893cb98a633daf8be6fdd286faa05c9044d/src/Illuminate/Database/Console/Migrations/StatusCommand.php#L79

### CODE REVIEW

**Comment by maximepvrt** [comment]
👀 @babenkoivan

File: src/Migrator.php [hunk 1]
```diff
@@ -92,7 +92,7 @@ public function rollbackAll(): self
         return $this;
     }
 
-    public function showStatus(): self
+    public function showStatus($onlyPending = false): self
```
Comment (babenkoivan) [inline src/Migrator.php:95]: Shall we make the output similar to Laravel's command? We can rewrite the method to something like this:

```php
public function showStatus(bool $onlyPending = false): self 
{
        $files = $this->migrationStorage->all();
        $migratedFiles = $this->migrationRepository->all();
        $lastBatch = $this->migrationRepository->lastBatch();
        
        $rows = $files->map(
            static fn (MigrationFile $file) => [
                $file->name(),
                $migratedFiles->contains($file->name()) 
                     ? '<fg=green;options=bold>Ran</>' . ($lastBatch->contains($file->name()) ? ' [last batch]'  : '') 
                     : '<fg=yellow;options=bold>Pending</>'
            ]
        )->when($onlyPending, static fn (Collection $rows) => $rows->filter(
            static fn (array $row) => strpos($row[1], 'Pending') !== false
        }));
        
        if ($rows->isNotEmpty()) {
            $this->newLine();
   
            $this->components->twoColumnDetail('<fg=gray>Migration name</>', '<fg=gray>Status</>');
        
            $rows->each(
                 fn ($migration) => $this->components->twoColumnDetail($rows[0], $rows[1])
             );
        
            $this->newLine();
        } elseif ($onlyPending) {
            $this->components->info('No pending migrations');
        } else {
            $this->components->info('No migrations found');
        }
}
```

I did not test this piece of code, but it looks more straightforward and similar to Laravel's implementation.

Comment (maximepvrt) [reply src/Migrator.php]: updated ;-)

Comment (maximepvrt) [reply src/Migrator.php]: @babenkoivan MR updated ;-)

**Comment by maximepvrt** [comment]
@babenkoivan PR updated ;-)

**Comment by babenkoivan** [comment]
Hey @maximepvrt, thank you for updating the PR! 

I've discovered that `Illuminate\Console\View\Components\Factory` is not supported by Laravel 6, and I don't want to lose compatibility because of such a small change. In addition, I forgot to ask you to update the tests and the documentation. 

I didn't want to make you rewrite the code again and wait until I had time to review it, so I made the changes myself (based on your code) and published a new version [v3.4.0](https://github.com/babenkoivan/elastic-migrations/releases/tag/v3.4.0), which includes `--pending` option for the status command 🙂

**Comment by maximepvrt** [comment]
rewrite the code is a good exercice 😝 but thanks 😊

**Pull request closed without merging**